### ISSUE TITLE
keyword/wildcard filtering

### ISSUE THREAD

**Issue by new1-test** [LinogeFly/hide-unwanted-streams-on-twitch#9]
Not an issue, but feature request.
because max amount of game titles are limited by max extension storage size, can you add feature to block all game titles that contain certain word or part of a word?

**Comment by LinogeFly** [comment]
Hello,

Thanks for your feedback. When I was developing this I certainly didn't expected ~~there is so much trash on Twitch~~ people would block that many channels/games so Chrome storage limit wouldn't be enough :).

But back to your suggestion. The thing is that I don't want to overcomplicate UI (user interface) in the extension. And this text pattern based feature would require additional UI elements like buttons, input fields, labels with explanation on how to specify test pattern etc. I don't really like that. I'm trying to keep UI simple and easy to use.

What I think I'm gonna do to solve this issue is I will add a compression for settings so it's possible to store more games/channels. I quickly checked compression options available and it looks pretty promising. For example, 180 KB of blocked items (about 3000 games/channels) become 4 KB which is just 2% of the original size. I will investigate more and try it out. I hope compression will solve the problem with the limit for extension settings in Chrome.

**Comment by new1-test** [comment]
you can add button that shows that additional ui elements on a different page.
~~and i think that perl syntax will be ok for wildcards, so no explanation will be needed~~(it's a joke).

one more justification for keyword blocking - entire game series blocking, because there always is someone, who's streaming old junk, so it clutters the directory page, taking your focus away from non seen titles.

**Comment by LinogeFly** [comment]
I agree that it should be a separate window for that but I still don't really like it :) I can't come up with a nice UI that I would like. I had similar suggestions some time ago that would require me to add more stuff to settings window. One was very similar to yours but it was about text in stream titles. For example, streams in French should be hidden if there are "[FRA]", "[FR]" or similar patterns in stream title.

I don't know, I will think about that. Such feature will add some flexibility to what kind of content you want to block but I just prefer to keep it more simple.

**Comment by Maverynthia** [comment]
HUSoT+?
I'd like to block whole blocks of games myself, instead of blocking each on individually.
